sources/httpUtils.ts

```diff
@@ -13,17 +13,25 @@ export async function fetchUrlStream(url: string, options: RequestOptions = {})
   const proxyAgent = new ProxyAgent();
 
   return new Promise<IncomingMessage>((resolve, reject) => {
-    const request = https.get(url, {...options, agent: proxyAgent}, response => {
-      const statusCode = response.statusCode;
-      if (statusCode != null && statusCode >= 200 && statusCode < 300)
-        return resolve(response);
+    const createRequest = (url: string) => {
+      const request: ClientRequest = https.get(url, {...options, agent: proxyAgent}, response => {
+        const statusCode = response.statusCode;
 
-      return reject(new Error(`Server answered with HTTP ${statusCode} when performing the request to ${url}; for troubleshooting help, see https://github.com/nodejs/corepack#troubleshooting`));
-    });
+        if ([301, 302, 307, 308].includes(statusCode as number))
```

Comment:
IIUC, adding a `Location` header is recommended, but not mandatory. We should check if it was indeed passed before trying to follow it
```suggestion
        if ([301, 302, 307, 308].includes(statusCode as number) && response.headers.location)
```

Reply:
Oh, I didn't consider that. Thank you very much for correcting me. 
Should I mention in the [Troubleshooting](https://github.com/nodejs/corepack#troubleshooting) section about the case of missing `Location` header during "redirection"?

Reply:
I'm not sure it would make much sense, if a proxy does a redirection without supplying a `Location` header, there's nothing the user can do about it, no? My guess is that in practice, proxies that do not define a `Location` header are very rare, and would feel broken for most users anyway.

Reply:
Yeah, you are right, they are very rare, and "Troubleshooting" should document some common issues.